evm-contracts/src/dev_v0.7/Operator.sol

```diff
@@ -0,0 +1,306 @@
+   * @dev Sets the LinkToken address for the imported LinkTokenInterface
+   * @param link The address of the LINK token
+   */
+  constructor(address link)
+    Owned()
+  {
+    s_LinkToken = LinkTokenInterface(link); // external but already deployed and unalterable
+  }
+
+  /**
+   * @notice Creates the Chainlink request
+   * @dev Stores the hash of the params as the on-chain commitment for the request.
+   * Emits OracleRequest event for the Chainlink node to detect.
+   * @param sender The sender of the request
+   * @param payment The amount of payment given (specified in wei)
+   * @param specId The Job Specification ID
+   * @param callbackAddress The callback address for the response
+   * @param callbackFunctionId The callback function ID for the response
+   * @param nonce The nonce sent by the requester
+   * @param dataVersion The specified data version
+   * @param data The CBOR payload of the request
+   */
+  function oracleRequest(
+    address sender,
+    uint256 payment,
+    bytes32 specId,
+    address callbackAddress,
+    bytes4 callbackFunctionId,
+    uint256 nonce,
+    uint256 dataVersion,
+    bytes calldata data
+  )
+    external
+    override
+    onlyLINK()
+    checkCallbackAddress(callbackAddress)
+  {
+    bytes32 requestId = keccak256(abi.encodePacked(sender, nonce));
+    require(s_commitments[requestId] == 0, "Must use a unique ID");
+    // solhint-disable-next-line not-rely-on-time
+    uint256 expiration = block.timestamp.add(EXPIRY_TIME);
+
+    s_commitments[requestId] = keccak256(
+      abi.encodePacked(
+        payment,
+        callbackAddress,
+        callbackFunctionId,
+        expiration
+      )
+    );
+
+    emit OracleRequest(
+      specId,
+      sender,
+      requestId,
+      payment,
+      callbackAddress,
+      callbackFunctionId,
+      expiration,
+      dataVersion,
+      data);
+  }
+
+  /**
+   * @notice Called by the Chainlink node to fulfill requests
+   * @dev Given params must hash back to the commitment stored from `oracleRequest`.
+   * Will call the callback address' callback function without bubbling up error
+   * checking in a `require` so that the node can get paid.
+   * @param requestId The fulfillment request ID that must match the requester's
+   * @param payment The payment amount that will be released for the oracle (specified in wei)
+   * @param callbackAddress The callback address to call for fulfillment
+   * @param callbackFunctionId The callback function ID to use for fulfillment
+   * @param expiration The expiration that the node should respond by before the requester can cancel
+   * @param data The data to return to the consuming contract
+   * @return Status if the external call was successful
+   */
+  function fulfillOracleRequest(
+    bytes32 requestId,
+    uint256 payment,
+    address callbackAddress,
+    bytes4 callbackFunctionId,
+    uint256 expiration,
+    bytes32 data
+  )
+    external
+    onlyAuthorizedNode
```

Comment:
We should make sure the order of modifiers fits Solidity's [style guide](https://solidity.readthedocs.io/en/latest/style-guide.html#function-declaration) recommendations:
1. Visibility
2. Mutability
3. Virtual
4. Override
5. Custom modifiers